boa/src/vm/mod.rs

```diff
@@ -259,26 +295,49 @@ impl Context {
                 let index = self.vm.read::<u32>();
                 let name = self.vm.frame().code.variables[index as usize].clone();
 
-                self.create_mutable_binding(name.as_ref(), false, VariableScope::Function)?;
+                if !self.has_binding(name.as_ref())? {
+                    self.create_mutable_binding(name.as_ref(), false, VariableScope::Function)?;
```

Comment:
I was looking at some failing tests and ended up comparing this to the interpreter and I think we're missing the step where we initialize variables with `undefined`, do you agree?
```suggestion
                    self.create_mutable_binding(name.as_ref(), false, VariableScope::Function)?;
                    self.initialize_binding(name.as_ref(), JsValue::undefined())?;
```

Reply:
Yes that's right. I already fixed that in my current local version. I think I will just push my current fixes again. Maybe we can just get to test parity with this PR.